code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/dao/impl/ServiceExposeMapDaoImpl.java

```diff
@@ -42,4 +66,95 @@ public Instance getServiceInstance(long serviceId, String instanceName) {
         }
         return instances.get(0);
     }
+
+    @Override
+    public Instance createServiceInstance(Map<Object, Object> properties, Service service, String instanceName) {
+        Map<String, Object> props = objectManager.convertToPropertiesFor(Instance.class,
+                properties);
+        final Instance instance = objectManager.create(Instance.class, props);
+            objectManager.create(ServiceExposeMap.class,
+                    SERVICE_EXPOSE_MAP.INSTANCE_ID,
+                    instance.getId(), SERVICE_EXPOSE_MAP.SERVICE_ID, service.getId());
+        DeferredUtils.nest(new Callable<Object>() {
```

Comment:
Darren requested to put it in. He wanted Create event to be generated before the process completes (and DeferredUtils.nest does it ); otherwise there were some delays in scheduling the start process.